doc/dev/custom_devices.rst

```diff
@@ -0,0 +1,39 @@
+
+Custom Devices
+##############
+
+Differences between Devices in PennyLane and Catalyst
+=====================================================
+
+PennyLane and Catalyst treat devices a bit differently.
+In PennyLane, one is able to `define devices <https://docs.pennylane.ai/en/stable/development/plugins.html>`_ in Python.
+Catalyst cannot interface with Python devices yet.
+Instead, Catalyst can only interact with devices that implement the `QuantumDevice <../api/file_runtime_include_QuantumDevice.hpp.html>`_ class.
+
+In addition to implementing the ``QuantumDevice`` class, one must implement the following method:
+
+.. code-block:: guess
+
+    extern "C" Catalyst::Runtime::QuantumDevice*
+    getCustomDevice() { return new CustomDevice(); }
+
+where ``CustomDevice()`` is a constructor for your custom device.
+``CustomDevice``'s destructor will be called by the runtime.
+
+.. note::
+
+    This interface might change quickly in the near future.
+    Please check back regularly for updates and to ensure your device is compatible with a specific version of Catalyst.
+
+I Have Implemented My Own QuantumDevice Class. How Do I Compile It?
+===================================================================
+
+There is an example ``CMakeLists.txt`` file and an example third party device in the runtime tests.
+Please take a look there.
```

Comment:
@erick-xanadu I recommend including links here, and providing a very brief overview/list of steps, including code snippets/directions (such as a basic custom device)